src/Mod/Path/PathScripts/features/PathFeatureExtensions.py

```diff
@@ -0,0 +1,305 @@
+                direction = self._getDirectedNormal(p0, normal)
+                if direction is None:
+                    return None
+
+                if PathGeom.pointsCoincide(normal, direction):
+                    r = circle.Radius - self.length.Value
+                else:
+                    r = circle.Radius + self.length.Value
+
+                # assuming the offset produces a valid circle - go for it
+                if r > 0:
+                    e3 = Part.makeCircle(r, circle.Center, circle.Axis, edge.FirstParameter * 180 / math.pi, edge.LastParameter * 180 / math.pi)
+                    if endPoints(edge):
+                        # need to construct the arc slice
+                        e0 = Part.makeLine(edge.valueAt(edge.FirstParameter), e3.valueAt(e3.FirstParameter))
+                        e2 = Part.makeLine(edge.valueAt(edge.LastParameter), e3.valueAt(e3.LastParameter))
+                        return Part.Wire([e0, edge, e2, e3])
+
+                    return Part.Wire([e3])
+
+                # the extension is bigger than the hole - so let's just cover the whole hole
+                if endPoints(edge):
+                    # if the resulting arc is smaller than the radius, create a pie slice
+                    PathLog.track()
+                    center = circle.Center
+                    e0 = Part.makeLine(center, edge.valueAt(edge.FirstParameter))
+                    e2 = Part.makeLine(edge.valueAt(edge.LastParameter), center)
+                    return Part.Wire([e0, edge, e2])
+
+                PathLog.track()
+                return Part.Wire([edge])
+
+            else:
+                PathLog.track(self.feature, self.sub, type(edge.Curve), endPoints(edge))
+                direction = self._getDirection(sub)
+                if direction is None:
+                    return None
+
+            #    return self._extendEdge(feature, edge, direction)
+            return self._extendEdge(feature, edges[0], direction)
+
+        return extendWire(feature, sub, self.length.Value)
+# Eclass
+
+
+def initialize_properties(obj):
+    """initialize_properties(obj)... Adds feature propeties to object argument"""
+    if not hasattr(obj, 'ExtensionLengthDefault'):
+        obj.addProperty('App::PropertyDistance', 'ExtensionLengthDefault', 'Extension', QtCore.QT_TRANSLATE_NOOP('PathPocketShape', 'Default length of extensions.'))
+    if not hasattr(obj, 'ExtensionFeature'):
+        obj.addProperty('App::PropertyLinkSubListGlobal', 'ExtensionFeature', 'Extension', QtCore.QT_TRANSLATE_NOOP('PathPocketShape', 'List of features to extend.'))
+    if not hasattr(obj, 'ExtensionCorners'):
+        obj.addProperty('App::PropertyBool', 'ExtensionCorners', 'Extension', QtCore.QT_TRANSLATE_NOOP('PathPocketShape', 'When enabled connected extension edges are combined to wires.'))
+        obj.ExtensionCorners = True
+
+    obj.setEditorMode('ExtensionFeature', 2)
+
+
+def set_default_property_values(obj, job):
```

Comment:
Everything in Path is camel case, please let's stick to that.

Reply:
My apologies.  I was attempting to adhere to conventions expressed at [Path Code Style Guide](https://forum.freecadweb.org/viewtopic.php?f=15&t=46845).  I will change them.

Thank you for the review.

Reply:
Done.